duckduckgo/Android
# Login detection for fireproofing

Task/Issue URL: https://app.asana.com/0/1149059203486286/1179242968971107/f
Tech Design URL: https://app.asana.com/0/1149059203486286/1176922255591995/f
CC: https://app.asana.com/0/1149059203486286/1176752473916553/f

**Description**:
Help users preserve logins to sites they frequent by adding the fireproofing prompt after login events on Android.

Includes:
- Login detection disable/enable from settings 
- Login detection core functionality 
- Login detection user prompt
- Pixels for login detection for fireproofing

![image](https://user-images.githubusercontent.com/4747865/85982231-c1e6d000-b9e5-11ea-8dc0-ebb629a64a14.png)

**Steps to test this PR**:
#Test 1: No login detection
1. Fresh install
1. Visit any site and log in
1. Ensure we don't ask to fireproof that website

#Test 2: Login detection
1. Go to Fireproof website settings
1. Enable login detection
1. Visit any site and log in (try twitter.com which uses its own domain for auth)
1. Ensure dialog ask to fireproof that website you have been forwarded to
1. Click on Fireproof
1. Ensure snackbar appears

#Test 3: Login detection
1. Go to Fireproof website settings
1. Enable login detection
1. Visit any site and log in (try wsj.com, user credentials inside Lastpass, different auth domain)
1. Ensure dialog ask to fireproof that website you have been forwarded to (wsj.com and not dowjones)
1. Click on Fireproof
1. Ensure snackbar appears
1. Additionally, you can clear data and check that the user keeps being signed in. (via menu -> My account)

#Test 4: pixel enable/disable login detection
1. Go to Fireproof website settings
1. Enable/Disable login detection
1. Ensure pixel is sent (enable: `m_fw_d_e` / disable: `m_fw_d_d`)

#Test 5: pixel on dialogs
1. With login detection enabled
1. Visit any site and log in
1. When dialog appears, ensure pixel sent: `m_fw_ld_s`
1. Click on "not now", ensure pixel sent: `m_fw_l_d`
1. Repeat and now click on "fireproof", ensure pixel sent: `m_fw_l_a`

#Test 6: pixel on overflow mane
1. Visit any site
1. If fireproof and remove fireproof to trigger pixels
1. When fireproof, pixel sent: `m_fw_a`
1. When removing fireproofing, pixel sent: `m_fw_r`

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
